contracts/contracts/rem.swap/src/rem.swap.cpp

```diff
@@ -0,0 +1,250 @@
+/**
+ *  @copyright defined in eos/LICENSE.txt
+ */
+
+#include <rem.swap/rem.swap.hpp>
+
+namespace eosio {
+
+    void swap::init( const name& user, public_key swap_key, const string& trx_id,
+                     asset& quantity, const time_point& swap_init_time ) {
+
+        require_auth( user );
+        swap_index swap_table( _self, get_first_receiver().value );
+
+        check( quantity.is_valid(), "invalid quantity" );
+        check( quantity.symbol == create_account_fee.symbol, "symbol precision mismatch" );
+        check( quantity.amount > create_account_fee.amount, "the quantity must be greater "
+                                                            "than the account creation fee" );
+
+        uint32_t timepoint = swap_init_time.sec_since_epoch();
+
+        std::vector<char> swap_payload( swap_key.data.begin(), swap_key.data.end() );
+        string payload_str = trx_id + quantity.to_string() + std::to_string(timepoint);
+        std::copy( payload_str.begin(), payload_str.end(), back_inserter(swap_payload));
+
+        checksum256 swap_hash = hash( swap_payload.data() );
```

Comment:
swap_id (or swap_hash) should sha256(trx_id+chain_id+swap_pub_key+quantity+swap_init_time)
not just public_key

Reply:
It's not a just public key, `std::copy` inserts all payload.

Reply:
Ok